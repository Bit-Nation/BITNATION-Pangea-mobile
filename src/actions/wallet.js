// @flow

import type { WalletType } from '../types/Wallet';

export type SelectWalletAction = { +type: 'SELECT_WALLET', +wallet: WalletType };
export type SendMoneyAction = { +type: 'SEND_MONEY', +amount: number, +toEthAddress: string };
export type WalletListUpdatedAction = { +type: 'WALLETS_LIST_UPDATED', +wallets: Array<WalletType> };
export type UpdateWalletListAction = { +type: 'UPDATE_WALLET_LIST' };
export type UpdateWalletBalanceAction = { +type: 'UPDATE_WALLET_BALANCE', +walletAddress: string, +walletCurrency: string};
export type SendMoneySuccessAction = { +type: 'SEND_MONEY_SUCCESS' };
export type SendMoneyFailedAction = { +type: 'SEND_MONEY_FAILED', +error: Error };
export type WalletSyncFailedAction = { +type: 'WALLET_SYNC_FAILED', +walletAddress: string, +walletCurrency: string, +error: Error };

export type Action =
  | SelectWalletAction
  | SendMoneyAction
  | WalletListUpdatedAction
  | UpdateWalletListAction
  | UpdateWalletBalanceAction
  | SendMoneySuccessAction
  | SendMoneyFailedAction
  | WalletSyncFailedAction;

export const SELECT_WALLET = 'SELECT_WALLET';
export const SEND_MONEY = 'SEND_MONEY';
export const SEND_MONEY_SUCCESS = 'SEND_MONEY_SUCCESS';
export const SEND_MONEY_FAILED = 'SEND_MONEY_FAILED';
export const WALLETS_LIST_UPDATED = 'WALLETS_LIST_UPDATED';
export const WALLET_SYNC_FAILED = 'WALLET_SYNC_FAILED';
export const UPDATE_WALLET_LIST = 'UPDATE_WALLET_LIST';
export const UPDATE_WALLET_BALANCE = 'UPDATE_WALLET_BALANCE';

/**
 * @desc Action creator for an action that select a wallet to be currently active.
 * @param {WalletType} wallet Wallet to select.
 * @returns {SelectWalletAction} An action.
 */
export function selectWallet(wallet: WalletType): SelectWalletAction {
  return {
    type: SELECT_WALLET,
    wallet,
  };
}

/**
 * @desc Action creator for an action that sends money.
 * @param {number} amount Amount to send in eth.
 * @param {string} toEthAddress Address to send money.
 * @returns {SendMoneyAction} An action.
 */
export function sendMoney(amount: number, toEthAddress: string): SendMoneyAction {
  return {
    type: SEND_MONEY,
    amount,
    toEthAddress,
  };
}

/**
 * @desc Action creator for an action that takes update of wallets available.
 * @param {WalletType[]} wallets Array of wallets.
 * @returns {WalletListUpdatedAction} An action.
 */
export function walletsListUpdated(wallets: Array<WalletType>): WalletListUpdatedAction {
  return {
    type: WALLETS_LIST_UPDATED,
    wallets,
  };
}

/**
 * @desc Action creator for an action that starts wallets list update.
 * @returns {UpdateWalletListAction} An action.
 */
export function updateWalletList(): UpdateWalletListAction {
  return {
    type: UPDATE_WALLET_LIST,
  };
}

/**
 * @desc Action creator for an action that starts specific wallet update.
 * @param {string} walletAddress Address of wallet to update.
 * @param {string} walletCurrency Currency of wallet to update.
 * @returns {UpdateWalletBalanceAction} An action.
 */
export function updateWalletBalance(walletAddress: string, walletCurrency: string): UpdateWalletBalanceAction {
  return {
    type: UPDATE_WALLET_BALANCE,
    walletAddress,
    walletCurrency,
  };
}

/**
 * @desc Action creator for an action that should be called on send money success.
 * @returns {SendMoneySuccessAction} An action.
 */
export function sendMoneySuccess(): SendMoneySuccessAction {
  return {
    type: SEND_MONEY_SUCCESS,
  };
}

/**
 * @desc Action creator for an action that should be called on send money failed.
 * @param {Error} error An error that occurs.
 * @returns {SendMoneyFailedAction} An action.
 */
export function sendMoneyFailed(error: Error): SendMoneyFailedAction {
  return {
    type: SEND_MONEY_FAILED,
    error,
  };
}

/**
 * @desc Action creator for an action that should be called on wallet synchronization fail.
 * @param {string} walletAddress Address of wallet that was synced.
 * @param {string} walletCurrency Current of wallet that was synced.
 * @param {Error} error An error that occurs.
 * @returns {WalletSyncFailedAction} An action.
 */
export function walletSyncFailed(walletAddress: string, walletCurrency: string, error: Error): WalletSyncFailedAction {
  return {
    type: WALLET_SYNC_FAILED,
    walletAddress,
    walletCurrency,
    error,
  };
}
