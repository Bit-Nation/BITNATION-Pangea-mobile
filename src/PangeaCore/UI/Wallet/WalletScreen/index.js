// @flow

import React from 'react';
import { View, WebView, TouchableHighlight } from 'react-native';
import Modal from 'react-native-modal';
import { connect } from 'react-redux';
import { Icon } from 'native-base';
import _ from 'lodash';
import Background from 'pangea-common-reactnative/UI/BackgroundImage';
import { screen } from 'pangea-common-reactnative/Screens';
import List from './List';
import EmptyWalletScreen from './EmptyState/index';
import { selectWallet, updateWalletList } from '@pangea/wallet/wallet-actions';
import NavigatorComponent from '../../NavigatorComponent';
import i18n from 'pangea-common/i18n';
import styles from '../../Nations/NationsScreen/NationsListScreen/styles';
import FakeNavigationBar from 'pangea-common-reactnative/UI/FakeNavigationBar';
import type { State as WalletState } from '@pangea/wallet/wallet-reducers';
import type { WalletType } from 'pangea-common/types/Wallet-types';
import type { Account } from 'pangea-common/types/accounts-types';
import type { Navigator } from 'pangea-common-reactnative/ReactNativeNavigation-types';
import { getCurrentAccount } from '@pangea/accounts/accounts-reducers';
import ScreenTitle from 'pangea-common-reactnative/UI/ScreenTitle';

const REMOVE_WALLETS_BUTTON = 'REMOVE_WALLETS_BUTTON';

type Props = {
  /**
   * @desc React Native Navigation navigator object.
   */
  navigator: Navigator,
  /**
   * @desc Current user object.
   */
  user: Account | null,
};

type TestingModeProps = {
  /**
   * @desc Flag that shows if testing mode is active.
   */
  testingModeActive: boolean,
  /**
   * @desc Function to remove all wallets.
   */
  removeWallets: () => void,
};

type Actions = {
  /**
   * @desc Function to select wallet.
   * @param {WalletType} wallet Wallet to be selected.
   */
  selectWallet: (wallet: WalletType) => void,
  /**
   * @desc Function to request wallet list update.
   */
  updateWalletList: () => void,
};

type State = {
  /**
   * @desc Flag to control Refreshing on Pull to Refresh
   */
  pullToRefreshInProgress: boolean,
  /**
   * @desc Stores Transaction Modal state
   */
  transactionModal: {
    /**
     * @desc Flag to control the visibility of modal
     */
    visible: boolean,
    /**
     * @desc Stores the ETH Address
     */
    ethAddress?: string,
  },
};

class WalletScreen extends NavigatorComponent<
  Props & TestingModeProps & Actions & WalletState,
  State,
> {
  constructor(props) {
    super(props);

    this.state = {
      pullToRefreshInProgress: false,
      transactionsVisible: false,
      ethAddress: '',
      transactionModal: {
        visible: false,
        ethAddress: '',
      },
    };
    this.props.updateWalletList();
  }

  componentWillReceiveProps(nextProps) {
    if (
      this.props.isRefreshing !== nextProps.isRefreshing &&
      nextProps.isRefreshing === false
    ) {
      this.setState({ pullToRefreshInProgress: false });
    }
  }

  onDidAppear() {
    this.props.updateWalletList();
  }

  onNavBarButtonPress(id) {
    if (id === REMOVE_WALLETS_BUTTON) {
      this.props.removeWallets();
    }
  }

  createWallet = () => {
    this.props.navigator.showModal(screen('CREATE_KEY_INTRODUCTION_SCREEN'));
  };

  restoreWallet = () => {};

  sendMoney = (wallet) => {
    this.props.selectWallet(wallet);
    this.props.navigator.push(screen('SEND_MONEY_SCREEN'));
  };

  receiveMoney = (wallet) => {
    this.props.selectWallet(wallet);
    this.props.navigator.push(screen('RECEIVE_MONEY_SCREEN'));
  };

  showTransactions = (wallet) => {
    this.setState({ transactionModal: { visible: true, ethAddress: wallet.ethAddress } });
  };

  onRefresh = () => {
    this.setState({ pullToRefreshInProgress: true });
    this.props.updateWalletList();
  };

  transactionModalClose() {
    this.setState({ transactionModal: { visible: false } });
  }

  render() {
    const { user } = this.props;
    return (
      <View style={styles.screenContainer}>
        <Background />
        <FakeNavigationBar />
        <View style={styles.bodyContainer}>
          <ScreenTitle title={i18n.t('screens.wallet.title')} />
          {this.props.wallets === null || _.isEmpty(this.props.wallets) ? (
            <EmptyWalletScreen
              onCreateWallet={this.createWallet}
              onRestoreWallet={this.restoreWallet}
            />
          ) : (
            <List
              wallets={this.props.wallets}
              onSendPress={this.sendMoney}
              onReceivePress={this.receiveMoney}
              onTransactionPress={this.showTransactions}
              onRefresh={this.onRefresh}
              isRefreshing={this.state.pullToRefreshInProgress}
            />
          )}
        </View>
        <Modal
          isVisible={this.state.transactionModal.visible}
        >
          <View>
            <TouchableHighlight
              onPress={() => {
                this.transactionModalClose();
              }}
            >
              <Icon name='ios-close' style={styles.closeIcon} />
            </TouchableHighlight>
          </View>
          {this.state.transactionModal && this.state.transactionModal.ethAddress &&
            <WebView
              source={user !== null && user.networkType === 'main' ? {
                uri: `https://etherscan.io/address/${
                  this.state.transactionModal.ethAddress
                  }`,
            } : {
              uri: `https://rinkeby.etherscan.io/address/${
                this.state.transactionModal.ethAddress
                  }`,
              }}
            />
          }
        </Modal>
      </View>
    );
  }
}

const mapStateToProps = state => ({
  ...state.wallet,
  testingModeActive: state.testingMode.isActive,
  user: getCurrentAccount(state.accounts),
});

const mapDispatchToProps = dispatch => ({
  selectWallet(wallet) {
    dispatch(selectWallet(wallet));
  },
  updateWalletList() {
    dispatch(updateWalletList());
  },
  removeWallets() {},
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(WalletScreen);
