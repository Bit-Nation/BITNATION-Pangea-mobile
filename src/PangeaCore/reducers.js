// @flow

import { combineReducers } from 'redux';
import accounts, { type State as AccountsState } from '@pangea/accounts/accounts-reducers';
import nations, { type State as NationsState } from '@pangea/nations/nations-reducers';
import modifyNation, { type State as ModifyNationState } from '@pangea/nations/modifyNation-reducers';
import wallet, { type State as WalletState } from '@pangea/wallet/wallet-reducers';
import key, { type State as KeyState } from '@pangea/key/key-reducers';
import activity, { type State as ActivityState } from '@pangea/activity/activity-reducers';
import testingMode, { type State as TestingModeState } from '@pangea/accounts/testingMode-reducers';
import chat, { type State as ChatState } from '@pangea/chat/chat-reducers';
import settings, { type State as SettingsState } from '@pangea/settings/settings-reducers';
import dApps, { type State as DAppsState } from '@pangea/dApps/dApps-reducers';
import documents, { type State as DocumentsState } from '@pangea/documents/documents-reducers';
import contacts, { type State as ContactsState } from '@pangea/contacts/contacts-reducers';

export type State = {
  accounts: AccountsState,
  nations: NationsState,
  modifyNation: ModifyNationState,
  wallet: WalletState,
  key: KeyState,
  activity: ActivityState,
  testingMode: TestingModeState,
  chat: ChatState,
  settings: SettingsState,
  dApps: DAppsState,
  documents: DocumentsState,
  contacts: ContactsState,
}

export const subReducers = {
  accounts,
  nations,
  modifyNation,
  wallet,
  key,
  activity,
  testingMode,
  chat,
  settings,
  dApps,
  documents,
  contacts,
};

export default combineReducers(subReducers);
