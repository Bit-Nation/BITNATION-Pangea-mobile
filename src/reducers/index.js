// @flow

import { combineReducers } from 'redux';
import accounts, { type State as AccountsState } from './accounts';
import nations, { type State as NationsState } from './nations';
import modifyNation, { type State as ModifyNationState } from './modifyNation';
import wallet, { type State as WalletState } from './wallet';
import key, { type State as KeyState } from './key';
import activity, { type State as ActivityState } from './activity';
import testingMode, { type State as TestingModeState } from './testingMode';
import chat, { type State as ChatState } from './chat';
import settings, { type State as SettingsState } from './settings';

export type State = {
  accounts: AccountsState,
  nations: NationsState,
  modifyNation: ModifyNationState,
  wallet: WalletState,
  key: KeyState,
  activity: ActivityState,
  testingMode: TestingModeState,
  chat: ChatState,
  settings: SettingsState,
}

export const subReducers = {
  accounts,
  nations,
  modifyNation,
  wallet,
  key,
  activity,
  testingMode,
  chat,
  settings,
};

export default combineReducers(subReducers);
