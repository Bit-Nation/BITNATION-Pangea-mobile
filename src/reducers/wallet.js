// @flow

import _ from 'lodash';

import {
  type Action,
  SELECT_WALLET,
  SEND_MONEY,
  SEND_MONEY_FAILED,
  SEND_MONEY_SUCCESS,
  UPDATE_WALLET_BALANCE,
  WALLET_SYNC_FAILED,
  WALLETS_LIST_UPDATED,
} from '../actions/wallet';
import type { WalletType } from '../types/Wallet';
import { getWalletIndex } from '../utils/wallet';

export type State = {
  +wallets: Array<WalletType> | null,
  +selectedWalletAddress: string | null,
  +moneySendingInProgress: boolean,
  +moneySendingError: Error | null,
};

export const initialState: State = {
  wallets: null,
  selectedWalletAddress: null,
  moneySendingInProgress: false,
  moneySendingError: null,
};

/**
 * @desc Wallet reducer.
 * @param {State} state Current state.
 * @param {Action} action Performed Action.
 * @returns {State} Next state.
 */
export default (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case SELECT_WALLET:
      return Object.assign({}, state, { selectedWalletAddress: action.wallet.currency });
    case WALLETS_LIST_UPDATED:
      return Object.assign({}, state, { wallets: _.cloneDeep(action.wallets) });
    case WALLET_SYNC_FAILED: {
      const { walletAddress } = action;
      const walletIndex = getWalletIndex(state.wallets || [], walletAddress);
      const newWallets = _.cloneDeep(state.wallets);
      if (walletIndex === null) {
        return state;
      }
      newWallets[walletIndex].synchronizationError = action.error;
      return Object.assign({}, state, { wallets: newWallets });
    }
    case UPDATE_WALLET_BALANCE: {
      const { walletAddress } = action;
      const walletIndex = getWalletIndex(state.wallets || [], walletAddress);
      if (walletIndex === null) {
        return state;
      }
      const newWallets = _.cloneDeep(state.wallets);
      newWallets[walletIndex].synchronizationError = undefined;
      return Object.assign({}, state, { wallets: newWallets });
    }
    case SEND_MONEY:
      return { ...state, moneySendingInProgress: true, moneySendingError: null };
    case SEND_MONEY_SUCCESS:
      return { ...state, moneySendingInProgress: false, moneySendingError: null };
    case SEND_MONEY_FAILED:
      return { ...state, moneySendingInProgress: false, moneySendingError: action.error };
    default:
      return state;
  }
};
