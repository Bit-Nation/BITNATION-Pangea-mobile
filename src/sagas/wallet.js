import { takeEvery } from 'redux-saga';

import { CREATE_WALLET, CREATE_MNEMONIC, createMnemonic } from '../actions/wallet';

async function createPrivateKey() {
  return 'privateKey_privateKey';
}

async function privateKeyToMnemonic(privateKey) {
  return ['word', 'word', 'word', 'word', 'word', 'word', 'word', 'word', 'word', 'word', 'word', 'word', 'word',
    'word', 'word', 'word', 'word', 'word', 'word', 'word', 'word', 'word', 'word', 'word'];
}

function* createWallet() {
  const privateKey = yield call(createPrivateKey);
  const mnemonic = yield call(privateKeyToMnemonic(privateKey));
  yield put(createMnemonic(mnemonic));
}

export default function* watchCreateWallet() {
  yield takeEvery(CREATE_WALLET, createWallet);
}
