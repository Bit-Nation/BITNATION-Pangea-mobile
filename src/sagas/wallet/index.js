// @flow

import {
  all,
  takeEvery,
} from 'redux-saga/effects';
import {
  SEND_MONEY,
  UPDATE_WALLET_BALANCE,
  UPDATE_WALLET_LIST,
} from '../../actions/wallet';
import {
  updateWalletList,
  updateWalletBalance,
  sendMoneySaga,
} from './sagas';
import { SERVICES_CREATED, SERVICES_DESTROYED } from '../../actions/serviceContainer';

/**
 * @desc Root wallet saga.
 * @return {void}
 */
export default function* rootSaga(): Generator<*, *, *> {
  yield all([
    takeEvery([UPDATE_WALLET_LIST, SERVICES_CREATED, SERVICES_DESTROYED], updateWalletList),
    takeEvery(UPDATE_WALLET_BALANCE, updateWalletBalance),
    takeEvery(SEND_MONEY, sendMoneySaga),
  ]);
}
