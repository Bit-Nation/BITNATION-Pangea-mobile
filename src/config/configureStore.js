// @flow

import { createStore, applyMiddleware, compose, type Store } from 'redux';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import createSagaMonitor from '@clarketm/saga-monitor';
import reducers from '../reducers';
import rootSaga from '../sagas';

/**
 * @desc Configures a Redux store.
 * @return {Store} Created store object.
 */
export default function configureStore(): Store {
  const sagaMonitor = createSagaMonitor({
    level: 'log',
    actionDispatch: true,
  });
  const sagaMiddleware = createSagaMiddleware({ sagaMonitor });

  const enhancer = compose(
    applyMiddleware(sagaMiddleware),
    applyMiddleware(logger),
  );

  const store = createStore(reducers, enhancer);
  sagaMiddleware.run(rootSaga);
  return store;
}
