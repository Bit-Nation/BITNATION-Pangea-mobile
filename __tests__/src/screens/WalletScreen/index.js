import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { initialState } from '../../../../src/reducers/wallet';
import WalletScreen from '../../../../src/screens/WalletScreen';
import navigatorMock from '../../../../__mocks__/Navigator';

describe('WalletScreen renders correctly', () => {
  const storeMock = configureStore([]);
  const propsMock = {
    navigator: navigatorMock,
  };

  test('empty', () => {
    const initialStateMock = {
      wallet: initialState,
      testingMode: {
        isActive: false,
      },
      accounts: {
        user: {},
      },
    };

    const wrapper = shallow((
      <WalletScreen {...propsMock} store={storeMock(initialStateMock)} />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  test('non-empty', () => {
    const initialStateMock = {
      wallet: {
        ...initialState,
        wallets: [{
          balance: '0',
          currency: 'ETH',
          ethAddress: '0xtestAddress',
        }],
      },
      testingMode: {
        isActive: false,
      },
      accounts: {
        user: {},
      },
    };

    const wrapper = shallow((
      <WalletScreen {...propsMock} store={storeMock(initialStateMock)} />
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
