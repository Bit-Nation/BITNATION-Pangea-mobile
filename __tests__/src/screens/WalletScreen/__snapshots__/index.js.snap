// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletScreen renders correctly empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(WalletScreen)
    navigator={
      Object {
        "pop": [MockFunction],
        "push": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "moneySendingError": null,
      "moneySendingInProgress": false,
      "moneySendingSuccess": false,
      "navigator": Object {
        "pop": [MockFunction],
        "push": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      },
      "removeWallets": [Function],
      "selectWallet": [Function],
      "selectedWalletAddress": null,
      "selectedWalletCurrency": null,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "testingModeActive": false,
      "updateWalletList": [Function],
      "wallets": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "moneySendingError": null,
        "moneySendingInProgress": false,
        "moneySendingSuccess": false,
        "navigator": Object {
          "pop": [MockFunction],
          "push": [MockFunction],
          "setButtons": [MockFunction],
          "setOnNavigatorEvent": [MockFunction],
        },
        "removeWallets": [Function],
        "selectWallet": [Function],
        "selectedWalletAddress": null,
        "selectedWalletCurrency": null,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "testingModeActive": false,
        "updateWalletList": [Function],
        "wallets": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WalletScreen renders correctly non-empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(WalletScreen)
    navigator={
      Object {
        "pop": [MockFunction],
        "push": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "moneySendingError": null,
      "moneySendingInProgress": false,
      "moneySendingSuccess": false,
      "navigator": Object {
        "pop": [MockFunction],
        "push": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      },
      "removeWallets": [Function],
      "selectWallet": [Function],
      "selectedWalletAddress": null,
      "selectedWalletCurrency": null,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "testingModeActive": false,
      "updateWalletList": [Function],
      "wallets": Array [
        Object {
          "balance": "0",
          "currency": "ETH",
          "ethAddress": "0xtestAddress",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "moneySendingError": null,
        "moneySendingInProgress": false,
        "moneySendingSuccess": false,
        "navigator": Object {
          "pop": [MockFunction],
          "push": [MockFunction],
          "setButtons": [MockFunction],
          "setOnNavigatorEvent": [MockFunction],
        },
        "removeWallets": [Function],
        "selectWallet": [Function],
        "selectedWalletAddress": null,
        "selectedWalletCurrency": null,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "testingModeActive": false,
        "updateWalletList": [Function],
        "wallets": Array [
          Object {
            "balance": "0",
            "currency": "ETH",
            "ethAddress": "0xtestAddress",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
